src/ensembl/src/shared/components/instant-download/instant-download-fetch/fetchForGene.ts

```diff
@@ -51,36 +51,41 @@ export const fetchForGene = async (payload: FetchPayload) => {
     options: { transcript: transcriptOptions, gene: geneOptions }
   } = payload;
 
-  const geneSequenceData = await fetchGeneSequenceMetadata({
-    genomeId,
-    geneId
-  });
-
-  const sequenceDownloadParams = geneSequenceData.transcripts.flatMap(
-    (transcript) =>
-      prepareDownloadParameters({
-        transcriptSequenceData: transcript,
-        options: transcriptOptions
-      })
-  );
+  try {
+    const geneSequenceData = await fetchGeneSequenceMetadata({
+      genomeId,
+      geneId
+    });
 
-  if (geneOptions.genomicSequence) {
-    sequenceDownloadParams.unshift(
-      getGenomicSequenceData(
-        geneSequenceData.stable_id,
-        geneSequenceData.unversioned_stable_id
-      )
+    const sequenceDownloadParams = geneSequenceData.transcripts.flatMap(
+      (transcript) =>
+        prepareDownloadParameters({
+          transcriptSequenceData: transcript,
+          options: transcriptOptions
+        })
     );
-  }
 
-  const worker = new SequenceFetcherWorker();
+    if (geneOptions.genomicSequence) {
+      sequenceDownloadParams.unshift(
+        getGenomicSequenceData(
+          geneSequenceData.stable_id,
+          geneSequenceData.unversioned_stable_id
+        )
+      );
+    }
 
-  const service = wrap<WorkerApi>(worker);
-  const sequences = await service.downloadSequences(sequenceDownloadParams);
+    const worker = new SequenceFetcherWorker();
 
-  worker.terminate();
+    const service = wrap<WorkerApi>(worker);
+    const sequences = await service.downloadSequences(sequenceDownloadParams);
 
-  downloadAsFile(sequences, `${geneId}.fasta`, {
-    type: 'text/x-fasta'
-  });
+    worker.terminate();
+
+    downloadAsFile(sequences, `${geneId}.fasta`, {
+      type: 'text/x-fasta'
+    });
+    return true;
```

Comment:
I am confused. What's the purpose of the returned boolean?

Reply:
It is used to check if the fetch was successful or not.